Сделал историю без ограничений и избавился от повторных просмотров истории,сохраняя последний просмотр

chunkychunk2 opened the pull request:
Усовершенствовал юнит тесты

Commit:
Сделал историю без ограничений,
Избавился от повторных просмотров истории,сохраняя последний просмотр,
Усовершенствовал юнит тесты

Commit:
Сделал историю без ограничений,
Избавился от повторных просмотров истории,сохраняя последний просмотр,
Усовершенствовал юнит тесты

Commit:
Сделал историю без ограничений,
Избавился от повторных просмотров истории,сохраняя последний просмотр,
Усовершенствовал юнит тесты

Commit:
Сделал историю без ограничений,
Избавился от повторных просмотров истории,сохраняя последний просмотр,
Усовершенствовал юнит тесты

Yang-Pi commented on src/main/java/com/yandex/taskmanager/HistoryManager.java:
```diff
@@ -0,0 +1,14 @@
+
+import java.util.List;
+
+public interface HistoryManager {
```
HistoryManager также должен содержать метод remove(id)

chunkychunk2 replied:
Добавил

Yang-Pi commented on src/main/java/com/yandex/taskmanager/service/InMemoryHistoryManager.java:
```diff
@@ -0,0 +1,27 @@
+
+import java.util.*;
+
+public class InMemoryHistoryManager implements HistoryManager {
```
В рамках класса InMemoryHistoryManager должны быть реализованы методы linkLast, getTasks, removeNode

chunkychunk2 replied:
Реализовал

Yang-Pi commented on src/main/java/com/yandex/taskmanager/service/InMemoryHistoryManager.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Override
+    public List<Task> getHistory() {
+        return new LinkedList<>(HISTORY.values());
```
Метод history() должен возвращать ArrayList. Суть всей задачи в том, чтобы самостоятельно реализовать связный список и понять как он работает, как устроены ссылки на начало и конец) А не использовать готовые решения стандартной библиотеки

chunkychunk2 replied:
Реализовал связанный список в классе Node

Yang-Pi reviewed (changes requested):
нужно реализовать свой вариант связного списка)

Commit:
реализовал методы linkLast, getTasks, removeNode
Добавил в HistoryManager метод remove(id) и реализовал его в InMemoryHistoryManager
Метод history() теперь возвращает ArrayList.
Создал класс Node с реализацией связанного списка

Yang-Pi commented on src/test/java/TaskTest.java:
```diff
@@ -190,4 +190,35 @@ void HistoryManagerTest() {
     }
+
+    @Test
+    void orderTest() {
```
Осталось написать несколько тестов, подтверждающих работу истории (можно написать как в Main, так и как @Test):

- удалите задачу, которая есть в истории, и проверьте, что при печати она не будет выводиться;
- удалите эпик с тремя подзадачами и убедитесь, что из истории удалился как сам эпик, так и все его подзадачи.

chunkychunk2 replied:
Добавил тесты, не была реализована логика удаления тасок из истории, добавил логику в последнем пуше

Yang-Pi reviewed (commented):
Отличная работа! Хорошая реализация с пустой головой и хвостом. Осталось написать несколько тестов

Commit:
добавил юнит тесты и проверки в main
реализовал логику удаления тасок из истории при удалении их через таскменеджер
